src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/wwwroot/css/app.css

```diff
@@ -62,3 +62,35 @@ a, .btn-link {
     .blazor-error-boundary::after {
         content: "An error has occurred."
     }
+
+.loading-progress {
+    position: relative;
+    display: block;
+    width: 8rem;
+    height: 8rem;
+    margin: 20vh auto 1rem auto;
+}
+
+    .loading-progress circle {
+        fill: none;
+        stroke: #e0e0e0;
+        stroke-width: 0.6rem;
+        transform-origin: 50% 50%;
+        transform: rotate(-90deg);
+    }
+
+        .loading-progress circle:last-child {
+            stroke: var(--blazor-load-percentage-color);
+            stroke-dasharray: calc(3.141 * var(--blazor-load-percentage) * 0.8), 500%;
+            transition: stroke-dasharray 0.05s ease-in-out;
+        }
+
+.loading-progress-text:after {
+    position: absolute;
+    content: var(--blazor-load-percentage-text);
+    color: black;
+    font-weight: bold;
+    text-align: center;
+    margin: -5.7rem 0 0 0.3rem;
+    left: 39.2vh;
+}
```

Comment:
Thank you for figuring out the correct positioning. I found that I was constantly modifying the `left` value because it was off-center every time that I tested it out, but I wasn't sure how to make it work. The method that you provided works well, thank you!